The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/10333336274/job/28605568186) [(plain)](https://github.com/rust-lang/rust/commit/1e4a02e31e8dd37731c010f1c372b7de6a24bdb7/checks/28605568186/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.700 Building wheels for collected packages: reuse
#13 2.701   Building wheel for reuse (pyproject.toml): started
#13 2.953   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 2.954   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132715 sha256=dfa09868353292d98f811d3efdb0d54d07389e808efc71d68e3b93c514bf8bec
#13 2.954   Stored in directory: /tmp/pip-ephem-wheel-cache-d6xuo_mi/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 2.956 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.363 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.363 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 3.885 Collecting virtualenv
#13 3.885 Collecting virtualenv
#13 3.924   Downloading virtualenv-20.26.3-py3-none-any.whl (5.7 MB)
#13 4.034      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.7/5.7 MB 52.9 MB/s eta 0:00:00
#13 4.075 Collecting distlib<1,>=0.3.7
#13 4.078   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#13 4.087      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 73.3 MB/s eta 0:00:00
#13 4.121 Collecting filelock<4,>=3.12.2
#13 4.125   Downloading filelock-3.15.4-py3-none-any.whl (16 kB)
#13 4.154 Collecting platformdirs<5,>=3.9.1
#13 4.157   Downloading platformdirs-4.2.2-py3-none-any.whl (18 kB)
#13 4.242 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.437 Successfully installed distlib-0.3.8 filelock-3.15.4 platformdirs-4.2.2 virtualenv-20.26.3
#13 DONE 4.5s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      212928 kB
DirectMap2M:     8175616 kB
DirectMap1G:    10485760 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/8291d68d926cedcdc77973e4c68f0828156d5bd8/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-8291d68d926cedcdc77973e4c68f0828156d5bd8-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
fmt check
Diff in /checkout/src/tools/compiletest/src/runtest.rs:3034:
             .stdout
             .lines()
             .filter(|line| line.starts_with(PREFIX))
-            .map(|line| line.replace(&self.testpaths.file.display().to_string(), "TEST_PATH").to_string())
+            .map(|line| {
+                line.replace(&self.testpaths.file.display().to_string(), "TEST_PATH").to_string()
+            })
             .map(|line| str_to_mono_item(&line, true))
             .collect();
 
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/compiler/rustc_target/src/spec/targets/armv7_unknown_freebsd.rs" "/checkout/compiler/rustc_target/src/spec/targets/armv5te_unknown_linux_uclibceabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/nvptx64_nvidia_cuda.rs" "/checkout/compiler/rustc_target/src/spec/targets/i686_unknown_openbsd.rs" "/checkout/compiler/rustc_target/src/spec/targets/armv7s_apple_ios.rs" "/checkout/compiler/rustc_target/src/spec/targets/i686_unknown_netbsd.rs" "/checkout/compiler/rustc_target/src/spec/targets/loongarch64_unknown_none_softfloat.rs" "/checkout/compiler/rustc_target/src/spec/targets/armv7_wrs_vxworks_eabihf.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_apple_ios_macabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/thumbv8m_main_none_eabihf.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_unknown_linux_musl.rs" "/checkout/compiler/rustc_target/src/spec/targets/i686_unknown_linux_gnu.rs" "/checkout/compiler/rustc_target/src/spec/targets/powerpc_unknown_freebsd.rs" "/checkout/compiler/rustc_target/src/spec/targets/loongarch64_unknown_none.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64h_apple_darwin.rs" "/checkout/compiler/rustc_target/src/spec/targets/mips64_openwrt_linux_musl.rs" "/checkout/compiler/rustc_target/src/spec/targets/thumbv7m_none_eabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/i686_uwp_windows_gnu.rs" "/checkout/compiler/rustc_target/src/spec/targets/armv4t_none_eabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/wasm64_unknown_unknown.rs" "/checkout/compiler/rustc_target/src/spec/targets/mips64_unknown_linux_muslabi64.rs" "/checkout/compiler/rustc_target/src/spec/targets/armv7k_apple_watchos.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_unknown_l4re_uclibc.rs" "/checkout/compiler/rustc_target/src/spec/targets/bpfeb_unknown_none.rs" "/checkout/compiler/rustc_target/src/spec/targets/arm64e_apple_darwin.rs" "/checkout/compiler/rustc_target/src/spec/targets/thumbv7a_pc_windows_msvc.rs" "/checkout/compiler/rustc_target/src/spec/targets/armv7_unknown_linux_musleabi.rs" "/checkout/src/tools/compiletest/src/header/test-auxillary/error_annotation.rs" "/checkout/compiler/rustc_target/src/spec/targets/i686_unknown_freebsd.rs" "/checkout/src/tools/compiletest/src/header/test-auxillary/known_legacy_directive.rs" "/checkout/compiler/rustc_target/src/spec/targets/powerpc_wrs_vxworks_spe.rs" "/checkout/src/tools/compiletest/src/header/test-auxillary/unknown_directive.rs" "/checkout/compiler/rustc_target/src/spec/targets/thumbv7m_nuttx_eabi.rs" "/checkout/src/tools/compiletest/src/header/test-auxillary/known_directive.rs" "/checkout/compiler/rustc_target/src/spec/targets/xtensa_esp32s3_none_elf.rs" "/checkout/compiler/rustc_target/src/spec/targets/riscv64gc_unknown_linux_musl.rs" "/checkout/compiler/rustc_target/src/spec/targets/i586_unknown_netbsd.rs" "/checkout/src/tools/compiletest/src/lib.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_apple_visionos_sim.rs" "/checkout/src/tools/compiletest/src/compute_diff.rs" "/checkout/compiler/rustc_target/src/spec/targets/mipsel_unknown_netbsd.rs" "/checkout/src/tools/compiletest/src/raise_fd_limit.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_unknown_fuchsia.rs" "/checkout/src/tools/compiletest/src/json.rs" "/checkout/compiler/rustc_target/src/spec/targets/riscv32im_unknown_none_elf.rs" "/checkout/src/tools/compiletest/src/tests.rs" "/checkout/compiler/rustc_target/src/spec/targets/armv7_unknown_netbsd_eabihf.rs" "/checkout/src/tools/compiletest/src/util.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_unknown_redox.rs" "/checkout/src/tools/compiletest/src/header.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_uwp_windows_gnu.rs" "/checkout/src/tools/compiletest/src/common.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_pc_windows_msvc.rs" "/checkout/src/tools/compiletest/src/errors.rs" "/checkout/compiler/rustc_target/src/spec/targets/thumbv7em_none_eabihf.rs" "/checkout/src/tools/compiletest/src/runtest.rs" "/checkout/compiler/rustc_target/src/spec/targets/hexagon_unknown_none_elf.rs" "/checkout/compiler/rustc_target/src/spec/targets/riscv32imac_esp_espidf.rs" "/checkout/compiler/rustc_target/src/spec/targets/riscv32gc_unknown_linux_musl.rs" "/checkout/compiler/rustc_target/src/spec/targets/s390x_unknown_linux_musl.rs" "/checkout/compiler/rustc_target/src/spec/targets/armv6k_nintendo_3ds.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_unknown_nto_qnx710.rs" "/checkout/compiler/rustc_target/src/spec/targets/wasm32_wasip2.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_apple_tvos.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Sat Aug 10 17:39:06 UTC 2024
  network time: Sat, 10 Aug 2024 17:39:06 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/src/tools/compiletest/src/runtest.rs
+++ b/src/tools/compiletest/src/runtest.rs
@@ -3027,11 +3027,17 @@ impl<'test> TestCx<'test> {
         const PREFIX: &str = "MONO_ITEM ";
         const CGU_MARKER: &str = "@@";
 
+        // Some MonoItems can contain {closure@/path/to/checkout/tests/codgen-units/test.rs}
+        // To prevent the current dir from leaking, we just replace the entire path to the test
+        // file with TEST_PATH.
         let actual: Vec<MonoItem> = proc_res
             .stdout
             .lines()
             .filter(|line| line.starts_with(PREFIX))
-            .map(|line| str_to_mono_item(line, true))
+            .map(|line| {
+                line.replace(&self.testpaths.file.display().to_string(), "TEST_PATH").to_string()
+            })
+            .map(|line| str_to_mono_item(&line, true))
             .collect();
 
         let expected: Vec<MonoItem> = errors::load_errors(&self.testpaths.file, None)

--- a/tests/codegen-units/item-collection/auxiliary/cgu_extern_closures.rs
+++ b/tests/codegen-units/item-collection/auxiliary/cgu_extern_closures.rs
@@ -1,3 +1,5 @@
+//@ compile-flags: -Zinline-mir=no
+
 #![crate_type = "lib"]
 
 #[inline]

--- a/tests/codegen-units/item-collection/cross-crate-closures.rs
+++ b/tests/codegen-units/item-collection/cross-crate-closures.rs
@@ -1,9 +1,6 @@
-// In the current version of the collector that still has to support
-// legacy-codegen, closures do not generate their own MonoItems, so we are
-// ignoring this test until MIR codegen has taken over completely
-//@ ignore-test
-
-//@ compile-flags:-Zprint-mono-items=eager
+// We need to disable MIR inlining in both this and its aux-build crate. The MIR inliner
+// will just inline everything into our start function if we let it. As it should.
+//@ compile-flags:-Zprint-mono-items=eager -Zinline-mir=no
 
 #![deny(dead_code)]
 #![feature(start)]
@@ -11,15 +8,15 @@
 //@ aux-build:cgu_extern_closures.rs
 extern crate cgu_extern_closures;
 
-//~ MONO_ITEM fn cross_crate_closures::start[0]
+//~ MONO_ITEM fn start @@ cross_crate_closures-cgu.0[Internal]
 #[start]
 fn start(_: isize, _: *const *const u8) -> isize {
-    //~ MONO_ITEM fn cgu_extern_closures::inlined_fn[0]
-    //~ MONO_ITEM fn cgu_extern_closures::inlined_fn[0]::{{closure}}[0]
+    //~ MONO_ITEM fn cgu_extern_closures::inlined_fn @@ cross_crate_closures-cgu.0[Internal]
+    //~ MONO_ITEM fn cgu_extern_closures::inlined_fn::{closure#0} @@ cross_crate_closures-cgu.0[Internal]
     let _ = cgu_extern_closures::inlined_fn(1, 2);
 
-    //~ MONO_ITEM fn cgu_extern_closures::inlined_fn_generic[0]<i32>
-    //~ MONO_ITEM fn cgu_extern_closures::inlined_fn_generic[0]::{{closure}}[0]<i32>
+    //~ MONO_ITEM fn cgu_extern_closures::inlined_fn_generic::<i32> @@ cross_crate_closures-cgu.0[Internal]
+    //~ MONO_ITEM fn cgu_extern_closures::inlined_fn_generic::<i32>::{closure#0} @@ cross_crate_closures-cgu.0[Internal]
     let _ = cgu_extern_closures::inlined_fn_generic(3, 4, 5i32);
 
     // Nothing should be generated for this call, we just link to the instance
@@ -28,5 +25,3 @@ fn start(_: isize, _: *const *const u8) -> isize {
 
     0
 }
-
-//~ MONO_ITEM drop-glue i8

--- a/tests/codegen-units/item-collection/non-generic-closures.rs
+++ b/tests/codegen-units/item-collection/non-generic-closures.rs
@@ -1,49 +1,45 @@
-// In the current version of the collector that still has to support
-// legacy-codegen, closures do not generate their own MonoItems, so we are
-// ignoring this test until MIR codegen has taken over completely
-//@ ignore-test
-
-//
-//@ compile-flags:-Zprint-mono-items=eager
+//@ compile-flags:-Zprint-mono-items=eager -Zinline-mir=no
 
 #![deny(dead_code)]
 #![feature(start)]
 
-//~ MONO_ITEM fn non_generic_closures::temporary[0]
+//~ MONO_ITEM fn temporary @@ non_generic_closures-cgu.0[Internal]
 fn temporary() {
-    //~ MONO_ITEM fn non_generic_closures::temporary[0]::{{closure}}[0]
+    //~ MONO_ITEM fn temporary::{closure#0} @@ non_generic_closures-cgu.0[Internal]
     (|a: u32| {
         let _ = a;
     })(4);
 }
 
-//~ MONO_ITEM fn non_generic_closures::assigned_to_variable_but_not_executed[0]
+//~ MONO_ITEM fn assigned_to_variable_but_not_executed @@ non_generic_closures-cgu.0[Internal]
 fn assigned_to_variable_but_not_executed() {
-    //~ MONO_ITEM fn non_generic_closures::assigned_to_variable_but_not_executed[0]::{{closure}}[0]
     let _x = |a: i16| {
         let _ = a + 1;
     };
 }
 
-//~ MONO_ITEM fn non_generic_closures::assigned_to_variable_executed_directly[0]
+//~ MONO_ITEM fn assigned_to_variable_executed_indirectly @@ non_generic_closures-cgu.0[Internal]
 fn assigned_to_variable_executed_indirectly() {
-    //~ MONO_ITEM fn non_generic_closures::assigned_to_variable_executed_directly[0]::{{closure}}[0]
+    //~ MONO_ITEM fn assigned_to_variable_executed_indirectly::{closure#0} @@ non_generic_closures-cgu.0[Internal]
+    //~ MONO_ITEM fn <{closure@TEST_PATH:27:13: 27:21} as std::ops::FnOnce<(i32,)>>::call_once - shim @@ non_generic_closures-cgu.0[Internal]
+    //~ MONO_ITEM fn <{closure@TEST_PATH:27:13: 27:21} as std::ops::FnOnce<(i32,)>>::call_once - shim(vtable) @@ non_generic_closures-cgu.0[Internal]
+    //~ MONO_ITEM fn std::ptr::drop_in_place::<{closure@TEST_PATH:27:13: 27:21}> - shim(None) @@ non_generic_closures-cgu.0[Internal]
     let f = |a: i32| {
         let _ = a + 2;
     };
     run_closure(&f);
 }
 
-//~ MONO_ITEM fn non_generic_closures::assigned_to_variable_executed_indirectly[0]
+//~ MONO_ITEM fn assigned_to_variable_executed_directly @@ non_generic_closures-cgu.0[Internal]
 fn assigned_to_variable_executed_directly() {
-    //~ MONO_ITEM fn non_generic_closures::assigned_to_variable_executed_indirectly[0]::{{closure}}[0]
+    //~ MONO_ITEM fn assigned_to_variable_executed_directly::{closure#0} @@ non_generic_closures-cgu.0[Internal]
     let f = |a: i64| {
         let _ = a + 3;
     };
     f(4);
 }
 
-//~ MONO_ITEM fn non_generic_closures::start[0]
+//~ MONO_ITEM fn start @@ non_generic_closures-cgu.0[Internal]
 #[start]
 fn start(_: isize, _: *const *const u8) -> isize {
     temporary();
@@ -54,7 +50,7 @@ fn start(_: isize, _: *const *const u8) -> isize {
     0
 }
 
-//~ MONO_ITEM fn non_generic_closures::run_closure[0]
+//~ MONO_ITEM fn run_closure @@ non_generic_closures-cgu.0[Internal]
 fn run_closure(f: &Fn(i32)) {
     f(3);
 }

--- a/tests/codegen-units/partitioning/methods-are-with-self-type.rs
+++ b/tests/codegen-units/partitioning/methods-are-with-self-type.rs
@@ -1,30 +1,23 @@
-// Currently, all generic functions are instantiated in each codegen unit that
-// uses them, even those not marked with #[inline], so this test does not make
-// much sense at the moment.
-//@ ignore-test
-
 // We specify incremental here because we want to test the partitioning for incremental compilation
 //@ incremental
 //@ compile-flags:-Zprint-mono-items=lazy
 
-#![allow(dead_code)]
-#![feature(start)]
+#![crate_type = "lib"]
 
-struct SomeType;
+pub struct SomeType;
 
 struct SomeGenericType<T1, T2>(T1, T2);
 
-mod mod1 {
+pub mod mod1 {
     use super::{SomeGenericType, SomeType};
 
     // Even though the impl is in `mod1`, the methods should end up in the
     // parent module, since that is where their self-type is.
     impl SomeType {
-        //~ MONO_ITEM fn methods_are_with_self_type::mod1[0]::{{impl}}[0]::method[0] @@ methods_are_with_self_type[External]
-        fn method(&self) {}
-
-        //~ MONO_ITEM fn methods_are_with_self_type::mod1[0]::{{impl}}[0]::associated_fn[0] @@ methods_are_with_self_type[External]
-        fn associated_fn() {}
+        //~ MONO_ITEM fn mod1::<impl SomeType>::method @@ methods_are_with_self_type[External]
+        pub fn method(&self) {}
+        //~ MONO_ITEM fn mod1::<impl SomeType>::associated_fn @@ methods_are_with_self_type[External]
+        pub fn associated_fn() {}
     }
 
     impl<T1, T2> SomeGenericType<T1, T2> {
@@ -52,25 +45,20 @@ mod type2 {
     pub struct Struct;
 }
 
-//~ MONO_ITEM fn methods_are_with_self_type::start[0]
-#[start]
-fn start(_: isize, _: *const *const u8) -> isize {
-    //~ MONO_ITEM fn methods_are_with_self_type::mod1[0]::{{impl}}[1]::method[0]<u32, u64> @@ methods_are_with_self_type.volatile[WeakODR]
+//~ MONO_ITEM fn start @@ methods_are_with_self_type[External]
+pub fn start() {
+    //~ MONO_ITEM fn mod1::<impl SomeGenericType<u32, u64>>::method @@ methods_are_with_self_type.volatile[External]
     SomeGenericType(0u32, 0u64).method();
-    //~ MONO_ITEM fn methods_are_with_self_type::mod1[0]::{{impl}}[1]::associated_fn[0]<char, &str> @@ methods_are_with_self_type.volatile[WeakODR]
+    //~ MONO_ITEM fn mod1::<impl SomeGenericType<char, &str>>::associated_fn @@ methods_are_with_self_type.volatile[External]
     SomeGenericType::associated_fn('c', "&str");
 
-    //~ MONO_ITEM fn methods_are_with_self_type::{{impl}}[0]::foo[0]<methods_are_with_self_type::type1[0]::Struct[0]> @@ methods_are_with_self_type-type1.volatile[WeakODR]
+    //~ MONO_ITEM fn <type1::Struct as Trait>::foo @@ methods_are_with_self_type-type1.volatile[External]
     type1::Struct.foo();
-    //~ MONO_ITEM fn methods_are_with_self_type::{{impl}}[0]::foo[0]<methods_are_with_self_type::type2[0]::Struct[0]> @@ methods_are_with_self_type-type2.volatile[WeakODR]
+    //~ MONO_ITEM fn <type2::Struct as Trait>::foo @@ methods_are_with_self_type-type2.volatile[External]
     type2::Struct.foo();
 
-    //~ MONO_ITEM fn methods_are_with_self_type::Trait[0]::default[0]<methods_are_with_self_type::type1[0]::Struct[0]> @@ methods_are_with_self_type-type1.volatile[WeakODR]
+    //~ MONO_ITEM fn <type1::Struct as Trait>::default @@ methods_are_with_self_type-type1.volatile[External]
     type1::Struct.default();
-    //~ MONO_ITEM fn methods_are_with_self_type::Trait[0]::default[0]<methods_are_with_self_type::type2[0]::Struct[0]> @@ methods_are_with_self_type-type2.volatile[WeakODR]
+    //~ MONO_ITEM fn <type2::Struct as Trait>::default @@ methods_are_with_self_type-type2.volatile[External]
     type2::Struct.default();
-
-    0
 }
-
-//~ MONO_ITEM drop-glue i8

